casper/src/main/scala/coop/rchain/casper/util/rholang/InterpreterUtil.scala

```diff
@@ -87,16 +89,14 @@ object InterpreterUtil {
       _         <- Span[F].mark("before-process-pre-state-hash")
       blockData = BlockData.fromBlock(block)
       isGenesis = block.header.parentsHashList.isEmpty
-      replayResult <- runtimeManager.replayComputeState(preStateHash)(
+      replayResult <- runtimeManager.replayComputeState(initialStateHash)(
                        internalDeploys,
                        internalSystemDeploys,
                        blockData,
                        invalidBlocks,
                        isGenesis
                      )
-      tsHash = ProtoUtil.postStateHash(block)
-      result <- handleErrors(tsHash, replayResult)
```

Comment:
@zsluedem Are you sure it's safe to remove this?

Reply:
Ah, it's called in `validateBlockCheckpoint` now, all good.

Reply:
I noticed that is it not called in `replayIntoMergeBlock` but looks like that is ok, because we are merging blocks that are already added to the DAG and are valid. 
So PR LGTM.